### PR TITLE
[nodeAnalyzer] Adding storage class support for runtime-scanner

### PR DESCRIPTION
**Pull request by mikescholl-sysdig**
## Checklist

[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]

- [x] Title of the PR starts with chart name (e.g. [mychartname])
- [x] Chart Version bumped for the respective charts
- [x] Changelog updated for the charts with version updates.
- [x] Variables are documented in the README.md (or README.tpl in some charts)
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers

Check Contribution guidelines in README.md for more insight.

### CODE REVIEW

**Comment by mikescholl-sysdig** [comment]
hey team; here is my test results:

```
sysdig-agent      admission-controller-scanner-664bf7bbbf-bb4bk                      1/1     Running    1 (18d ago)    18d
sysdig-agent      admission-controller-webhook-699dfd7495-mzwzr                      1/1     Running    0              2d5h
sysdig-agent      admission-controller-webhook-699dfd7495-wbnd8                      1/1     Running    0              18d
sysdig-agent      sysdig-agent-5dmgs                                                 0/1     Init:0/1   0              3s
sysdig-agent      sysdig-agent-fj6xd                                                 0/1     Init:0/1   0              3s
sysdig-agent      sysdig-agent-kspmcollector-696fcdf48-5wwx7                         1/1     Running    0              4d18h
sysdig-agent      sysdig-agent-ncx7n                                                 0/1     Init:0/1   0              3s
sysdig-agent      sysdig-agent-node-analyzer-54d5f                                   5/5     Running    0              4m23s
sysdig-agent      sysdig-agent-node-analyzer-nwnbx                                   5/5     Running    0              4m43s
sysdig-agent      sysdig-agent-node-analyzer-rhgl8                                   5/5     Running    0              3m55s
sysdig-agent      sysdig-agent-rapidresponse-8lg29                                   1/1     Running    0              4d18h
sysdig-agent      sysdig-agent-rapidresponse-bjph6                                   0/1     Pending    0              29s
sysdig-agent      sysdig-agent-rapidresponse-lt44r                                   1/1     Running    0              4d18h
❯ kubectl get pvc -A
NAMESPACE      NAME                                       STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
sysdig-agent   sysdig-agent-node-analyzer-54d5f-tmp-vol   Bound    pvc-19f1376e-fb42-48dd-9b60-6034863d6ebc   4Gi        RWO            aws-gp2        4m35s
sysdig-agent   sysdig-agent-node-analyzer-nwnbx-tmp-vol   Bound    pvc-bdd84ba7-f039-4d66-9abb-465c8618c354   4Gi        RWO            aws-gp2        5m
sysdig-agent   sysdig-agent-node-analyzer-rhgl8-tmp-vol   Bound    pvc-596efe12-17eb-45d6-b629-6b13c1f8f69a   4Gi        RWO            aws-gp2        4m7s
```
My definition of my storageClass:
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: aws-gp2
provisioner: kubernetes.io/aws-ebs
parameters:
  type: gp2
reclaimPolicy: Delete
volumeBindingMode: Immediate
```

and my sysdig-deploy values.yaml
```
global:
  clusterConfig:
    name: sysdig-flatcar-cluster-1
  sysdig:
    accessKeySecret: sysdig-agent-secret
    region: us2
  image:
    registry: quay.io
  kspm:
    deploy: true
  VerifyCertificate: false
agent:
  daemonset:
    updateStrategy:
      type: RollingUpdate
    env: {}
    nodeSelector: {}
    affinity: {}
    annotations: {}
    probes:
      initialDelay: 60
  ebpf:
    enabled: true
    settings:
      mountEtcVolume: true
  resources:
    requests: 
      cpu: 250m
      memory: 512Mi
    limits:
      cpu: 1000m
      memory: 1Gi
  leaderelection:
    enable: false
  priorityClassName: system-node-critical
  psp:
    create: true
  rbac:
    create: true
  resourceProfile: custom
  scc:
    create: true
  secure:
    enabled: true
    vulnerabilityManagement:
      newEngineOnly: false
  serviceAccount:
    create: true
    name: sysdig-agent
  slim:
    enabled: true
    image:
      repository: sysdig/agent-slim
    kmoduleImage:
      repository: sysdig/agent-kmodule
      tag: 12.8.1
    resources:
      requests:
        cpu: 100m
        memory: 32Mi
      limits:
        memory: 512Mi
        cpu: 500m
  sysdig:
    settings:
      collector: ingest-us2.app.sysdig.com
      collector_port: 6443
      log:
        file_priority: debug
        console_priority: debug
      watchdog:
        max_memory_usage_mb: 1024
        max_memory_usage_subprocesses:
          sdchecks: 128
          sdjagent: 256
          mountedfs_reader: 32
          statsite_forwarder: 32
          cointerface: 512
          promscrape: 640
  tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
  - key: CriticalAddonsOnly
    operator: Equal
    value: 'true'
    effect: NoSchedule
nodeAnalyzer:
  image:
    pullPolicy: Always
  nodeAnalyzer:
    deploy: true
    apiEndpoint: us2.app.sysdig.com
    extraVolumes: {}
    debug: false
    tolerations:
    - effect: NoSchedule
      key: node-role.kubernetes.io/master
    - effect: NoSchedule
      key: node-role.kubernetes.io/control-plane
    - key: CriticalAddonsOnly
      operator: Equal
      value: 'true'
      effect: NoSchedule
    priorityClassName: system-node-critical
    hostAnalyzer:
      env:
        ENABLE_FLATCAR_SCAN: true
    kspmAnalyzer:
      deploy: true
      image:
        repository: sysdig/kspm-analyzer
        pullPolicy: IfNotPresent
    runtimeScanner:
      deploy: true
      storageClassName: aws-gp2
      settings:
        eveEnabled: true
  psp:
    create: true
  rbac:
    create: true
  scc:
    create: true
  secure:
    enabled: true
    vulnerabilityManagement:
      newEngineOnly: false
rapidResponse:
  enabled: true
  rapidResponse:
    passphrase: "<redacted>"
    resources:
      requests:
        cpu: 100m
        memory: 256Mi
      limits:
        cpu: 500m
        memory: 512Mi

**Comment by mikescholl-sysdig** [comment]
updating my `ephemeral-storage` requests and limits and it recreated the volumes as expected:

```
    runtimeScanner:
      deploy: true
      storageClassName: aws-gp2
      settings:
        eveEnabled: true
      resources:
        requests:
          ephemeral-storage: 10Gi
        limits:
          ephemeral-storage: 10Gi
```

```
❯ helm upgrade sysdig-agent --namespace sysdig-agent ../../../../helm-charts/sysdig-charts/charts/sysdig-deploy --values ./sysdig-deploy-flatcar-values.yml
Release "sysdig-agent" has been upgraded. Happy Helming!
NAME: sysdig-agent
LAST DEPLOYED: Wed Sep 21 13:09:01 2022
NAMESPACE: sysdig-agent
STATUS: deployed
REVISION: 17
NOTES:
The agent for Sysdig Secure DevOps Platform is spinning up on each node in your
cluster. After a few seconds, you should see your hosts appearing in the
Sysdig Agent Health & Status Dashboard.

Links for your deployment:
  * Sysdig Monitor: https://us2.app.sysdig.com/#/dashboard-template/view.sysdig.agents?last=10
  * Sysdig Secure: https://us2.app.sysdig.com/secure/#/events/l:600/*/*?viewAs=list
❯ kubectl get pv -A
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS     CLAIM                                                   STORAGECLASS   REASON   AGE
pvc-19f1376e-fb42-48dd-9b60-6034863d6ebc   4Gi        RWO            Delete           Released   sysdig-agent/sysdig-agent-node-analyzer-54d5f-tmp-vol   aws-gp2                 32m
pvc-596efe12-17eb-45d6-b629-6b13c1f8f69a   4Gi        RWO            Delete           Bound      sysdig-agent/sysdig-agent-node-analyzer-rhgl8-tmp-vol   aws-gp2                 31m
pvc-9d9f8732-468c-46f2-9603-5defd3acced2   10Gi       RWO            Delete           Bound      sysdig-agent/sysdig-agent-node-analyzer-9s9cf-tmp-vol   aws-gp2                 2s
pvc-bdd84ba7-f039-4d66-9abb-465c8618c354   4Gi        RWO            Delete           Bound      sysdig-agent/sysdig-agent-node-analyzer-nwnbx-tmp-vol   aws-gp2                 32m
❯ kubectl get pv -n sysdig-agent
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                                                   STORAGECLASS   REASON   AGE
pvc-596efe12-17eb-45d6-b629-6b13c1f8f69a   4Gi        RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-rhgl8-tmp-vol   aws-gp2                 31m
pvc-9d9f8732-468c-46f2-9603-5defd3acced2   10Gi       RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-9s9cf-tmp-vol   aws-gp2                 17s
pvc-bdd84ba7-f039-4d66-9abb-465c8618c354   4Gi        RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-nwnbx-tmp-vol   aws-gp2                 32m
❯ kubectl get pv -n sysdig-agent
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                                                   STORAGECLASS   REASON   AGE
pvc-484c6183-429a-4ee0-93f1-4a41d98ff893   10Gi       RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-v7tnq-tmp-vol   aws-gp2                 44s
pvc-9d9f8732-468c-46f2-9603-5defd3acced2   10Gi       RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-9s9cf-tmp-vol   aws-gp2                 81s
pvc-dcb1337d-1fc4-4e7f-9705-0c6d56a23dbc   10Gi       RWO            Delete           Bound    sysdig-agent/sysdig-agent-node-analyzer-h5ptn-tmp-vol   aws-gp2                 56s
```

File: charts/node-analyzer/values.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ image:
   # Secrets must be manually created in the namespace.
   # ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
   #
-  # pullSecrets:
+  pullSecrets: {}
```
Comment (guidobonomi) [inline charts/node-analyzer/values.yaml:29]: changes on the pullSecrets here seems out of scope from this PR. May I ask why do we want to change these?

File: charts/node-analyzer/templates/daemonset-node-analyzer.yaml [hunk 2]
```diff
@@ -63,7 +63,21 @@ spec:
...
+          ephemeral:
+            volumeClaimTemplate:
+              metadata:
+                labels:
+                  type: {{ .Values.nodeAnalyzer.runtimeScanner.storageClassName }}
+              spec:
+                accessModes: [ "ReadWriteOnce" ]
+                storageClassName: {{ .Values.nodeAnalyzer.runtimeScanner.storageClassName }}
+                resources:
+                  requests:
```
Comment (guidobonomi) [inline charts/node-analyzer/templates/daemonset-node-analyzer.yaml:76]: we're mapping limits to requests and we're not including limits as key here. Is that made on purpose?

File: charts/node-analyzer/README.md [hunk 3]
```diff
@@ -141,6 +148,7 @@ The following table lists the configurable parameters of the Sysdig Node Analyze
 | `nodeAnalyzer.benchmarkRunner.env`                                   | Extra environment variables that will be passed onto pods                                | `{}`                                                                           |
 | `nodeAnalyzer.runtimeScanner.deploy`                                 | Deploy the Runtime Scanner                                                               | `false`                                                                        |
 | `nodeAnalyzer.runtimeScanner.extraMounts`                            | Specify a container engine custom socket path (docker, containerd, CRI-O)                |                                                                                |
+| `nodeAnalyzer.runtimeScanner.storageClassName`                 | Runtime Scanner storage class to use instead of emptyDir for ephemeral storage                                                    | ``                                                                         |
```
Comment (yashgiri) [inline charts/node-analyzer/README.md]: Not necessary to modify but small nit on indentation

**Review by yashgiri — CHANGES_REQUESTED** [review]
Can you please also include a version change for sysdig-deploy and a changelog for sysdig-deploy indicating NA version bump.

File: charts/node-analyzer/values.yaml (hunk 1 above)
Comment (mikescholl-sysdig) [reply charts/node-analyzer/values.yaml:29]: @guidobonomi when testing locally the system complained about these not being included in the values.yaml. Generally if you have a field you don't want to have populated in a values.yaml best practice is to leave the field with a empty set rather than uncomment it as it will be potentially excluded from auto-documentation methods.

File: charts/node-analyzer/templates/daemonset-node-analyzer.yaml (hunk 2 above)
Comment (mikescholl-sysdig) [reply charts/node-analyzer/templates/daemonset-node-analyzer.yaml:76]: @guidobonomi i have updated this. It should be referencing the proper req/limits now. Please feel free to re-review

File: charts/sysdig-deploy/ci/test-values.yaml.template [hunk 4]
```diff
@@ -3,3 +3,25 @@ global:
     namespace: ""
   sysdig:
     accessKey: ${SECURE_AGENT_TOKEN}
+agent:
```
Comment (yashgiri) [inline charts/sysdig-deploy/ci/test-values.yaml.template]: all the changes in these files can be removed now

File: charts/sysdig-deploy/CHANGELOG.md [hunk 5]
```diff
@@ -4,8 +4,11 @@
 
 This file documents all notable changes to Sysdig's sysdig-deploy Helm Chart. The release numbering uses [semantic versioning](http://semver.org).
 
+## v1.3.22
+### Minor Changes
+* node-analyzer bumping version to ~1.7.19
+
 ## v1.3.21
-### Minor changes
```
Comment (yashgiri) [inline charts/sysdig-deploy/CHANGELOG.md:8]: why remove this ?

**Review by yashgiri — CHANGES_REQUESTED** [review]
Only a small nit (not too important) but otherwise changes look good to me.

**Review by yashgiri — APPROVED** [review]
